src/adapters/poi/geocoder_poi.js

```diff
@@ -0,0 +1,82 @@
+import Poi from "./poi";
+
+export default class GeocoderPoi extends Poi {
+  constructor(feature) {
+    let poiClassText = ''
+    let poiSubclassText = ''
+
+    if (feature.properties.geocoding.properties && feature.properties.geocoding.properties.length > 0) {
+      let poiClass = feature.properties.geocoding.properties.find((property) => {
+        return property.key === 'poi_class'
+      })
+
+      if (poiClass) {
+        poiClassText = poiClass.value
+      }
+      let poiSubclass = feature.properties.geocoding.properties.find((property) => {
+        return property.key === 'poi_subclass'
+      })
+      if (poiSubclass) {
+        poiSubclassText = poiSubclass.value
+      }
+    }
+    let addressLabel = ''
+    if (feature.properties && feature.properties.geocoding && feature.properties.geocoding.address) {
+      addressLabel = feature.properties.geocoding.address.label
+    }
+
+    /* generate name corresponding to poi type */
+    let name = ''
+    let adminLabel = ''
+    const resultType = feature.properties.geocoding.type
+    switch (resultType) {
+      case 'poi':
+        name = feature.properties.geocoding.name
+        break
+      case 'house':
+        name = feature.properties.geocoding.name
+        break
+      case 'street':
+        name = feature.properties.geocoding.name
+        break
+      default: /* admin */
+        let splitPosition = feature.properties.geocoding.label.indexOf(',')
+        let nameFragments
+        if (splitPosition === -1) {
+          nameFragments = [feature.properties.geocoding.label]
+        } else {
+          nameFragments = [feature.properties.geocoding.label.slice(0, splitPosition), feature.properties.geocoding.label.slice(splitPosition + 1)]
+        }
+        if (nameFragments.length > 1) {
+          name = nameFragments[0]
+          adminLabel = nameFragments[1]
+        } else {
+          name = feature.properties.geocoding.label
+        }
+    }
+
+    super(feature.properties.geocoding.id, name, resultType, {
+      lat: feature.geometry.coordinates[1],
+      lng: feature.geometry.coordinates[0]
+    }, poiClassText, poiSubclassText)
+    /* extract custom data for autocomplete */
+    this.value = feature.properties.geocoding.label
+    this.addressLabel = addressLabel
+    this.adminLabel = adminLabel
+
+    this.postcode = feature.properties.geocoding.postcode
```

Comment:
The geocoder may return multiple postcodes (separated by `;`)
Potentially many postcodes (a street in Paris, with no associated borough will contained 20 postcodes !)

Should we take the first one ?

Reply:
i'll stop on the first one